core.c

```diff
@@ -39,7 +39,7 @@ static void serial_tx(struct em8051 *aCPU) {
 	       return;
 
 	aCPU->serial_out_remaining_bits--;
-	int tx_bit = (aCPU->mSFR[REG_SBUF] >> aCPU->serial_out_remaining_bits) & 0x01;
+	uint8_t tx_bit = (aCPU->mSFR[REG_SBUF] >> aCPU->serial_out_remaining_bits) & 0x01;
```

Comment:
how do you feel about stdbool.h for bits, instead of uint8_t?

Reply:
Good idea in my book, but it seems we are slowly going towards C99 ... 🤣

@jarikomppa what do you think about it ? 

Since one of the design trait of this project is to aim for "ANSI C for maximum compatibility". That said, for me, C99 compliance is the absolute minimum nowadays.

Reply:
Ok, did that in 0e147099f66bac41152a1885b3ff21b7ce05ab3c and a7ee2354079d4d02ecc404ac5187baa9f230991a.
It actually broke, and I had to fix it in f1e75052d79509b8a43eb69c757eea9cedf0f2d9.

That fix also contains a change of naming, as I didn't understood what the code did at first. Now it looks obvious.